### ISSUE TITLE
Build dedicated interfaces for PIO on x86 platforms

### ISSUE THREAD

**Issue by jiangliu** [rust-vmm/vm-device#22]
X86 systems supports both MMIO and PIO, most other systems only support MMIO.
To use unify interfaces to support both MMIO and PIO on x86 systems, some enums are introduced:
```rust
/// IO Addresses.
#[derive(Debug, Copy, Clone, Deserialize, Serialize)]
pub enum IoAddress {
    /// Port I/O address.
    Pio(u16),

    /// Memory mapped I/O address.
    Mmio(u64),
}
```

This design causes much extra works to driver implementations and unit test cases. Even on the vCPU thread, we need to encapsulate PIO address and MMIO address into IoAddress. It works as:
1. vCPU threads create IoAddress from PIO/MMIO address
2. IoManager looks up device driver by IoAddress 
3. device driver decode IoAddress into PIO/MMIO address.

It also needs more storage/register to store IoAddress than Pio/MMIO address.

On the other hand, DeviceManager already has dedicated interfaces for PIO and MMIO, so suggest to build dedicated interfaces for PIO in DeviceIo, and make PIO related interfaces as conditional compilation for x86 only.

**Comment by bonzini** [comment]
In QEMU, MMIO and PIO are treated simply as two instances of the same thing (the "address space" which, unlike `vm-memory`'s GuestRegionMmap, also includes I/O addresses). This makes it very easy to support the same device as both PIO and MMIO, for example the serial port. I think we should avoid enums in favor of having multiple instances of the manager objects (composition).

**Comment by andreeaflorescu** [comment]
This proposal looks good to me. @jiangliu do you want to send a PR with the changes?

**Comment by jiangliu** [comment]
> This proposal looks good to me. @jiangliu do you want to send a PR with the changes?

Sure, I will send a PR for it:)

**Comment by jiangliu** [comment]
There's one question. IO port is u16 on x86, but it's u32 with  IA64 and PCI bus. Is it u32 on ARM64?

**Comment by andreeaflorescu** [comment]
I really don't know how this works on arm64. I can try to ping some people that might know more about this. Can we go with a u32 and adjust this afterwards?

**Comment by bonzini** [comment]
Yes, we can use u32 or even u64.

**Comment by andreeaflorescu** [comment]
Fixed by: #25
